### PR TITLE
Keep user legal reasons

### PR DESCRIPTION
**Pull request by christalib**
* [X] Are you working on the latest release?
* [X] Have you tested it locally?
* [X] Please explain your change

Previously there was the possibility for user to delete their account
and to have their contributions linked to an anonymous account.

However, after discussion with our lawyer, we cannot sustain this
solution as we must be able to keep a certain amount of data for legal
reasons.

Currently this, commit gives the possibility for users to cancel their
subscribtion to the website and adds a "[USER LEFT}" to their username
so other people know this user isn't active anymore (a little bit like
Wikipedia).

I'm open to other solutions such as scrambling the username for example.
Mind that usernames must be unique so this could lead to certain issues.

**As this change is quite big in how we operate, I'm requesting reviews here.**

### CODE REVIEW

**Comment by christalib** [comment]
@michielbdejong @madoleary Hi! Could I have a review on this? It's an important thing we discussed during the Berlin sprint.

**Comment by christalib** [comment]
Merge rebased and repushed. Anyone?

File: app/views/users/registrations/edit.html.erb [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
     <h3>Cancel my account</h3>
 
-    <p>Unhappy? <%= link_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure? We will delete your personal information however we keep your contributions and we assign them to another anonymous account." }, method: :delete %></p>
+    <p>Unhappy? <%= link_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure? We will cancel your access to the website however your contributions will still be displayed, a little bit like for the same reasons than Wikipedia." }, method: :delete %></p>
```
Comment (michielbdejong) [inline app/views/users/registrations/edit.html.erb]: 'the same reasons than' -> 'the same reasons as'
isn't 'a bit like for' an anacoluthon? maybe we can just say:
> Are you sure? We will cancel your access to the website, however your contributions will still be displayed, a little bit like Wikipedia's account deactivation process.

and then maybe add a link to https://en.wikipedia.org/wiki/Wikipedia:FAQ#How_do_I_change_my_username/delete_my_account?

Comment (christalib) [reply app/views/users/registrations/edit.html.erb]: Yep, sure!

**Comment by christalib** [comment]
> LGTM, especially if this aligns us to how wikipedia does it too.
> Did you test this and is it working?

I have tested it at the time I did the change. I wouldn't mind trying it out in a staging environment with the current code base. I haven't followed the development since then, has a lot of thing changed database-wise?

**Pull request closed without merging**